src/utils/c_utils/src/string_c_utils.c

```diff
@@ -453,7 +453,7 @@ int ADUC_AllocAndStrCopyN(char** dest, const char* src, size_t srcLen)
     size_t num_copied = 0;
     char* tmpDest = NULL;
 
-    if (dest == NULL || srcLen == 0)
+    if (dest == NULL || src == NULL || srcLen == 0)
```

Comment:
return values differ from your ADUC_Safe_StrCopyN which returns number of chars copied.  i can see this potentially causing coding errors.

Reply:
they both now return int where 0 is success.